api4/config_test.go

```diff
@@ -63,8 +63,8 @@ func TestGetConfigWithAccessTag(t *testing.T) {
 	th := Setup(t)
 	defer th.TearDown()
 
-	varyByHeader := *&th.App.Config().RateLimitSettings.VaryByHeader // environment perm.
-	supportEmail := *&th.App.Config().SupportSettings.SupportEmail   // site perm.
+	varyByHeader := th.App.Config().RateLimitSettings.VaryByHeader // environment perm.
+	supportEmail := th.App.Config().SupportSettings.SupportEmail   // site perm.
```

Comment:
This is incorrect. `SupportEmail` is a pointer. We should dereference that, and use the pointer to the variable in line 70. 

Actually the entire `defer th.App.UpdateConfig` part is unnecessary. We can remove it.

Reply:
I do't understand the `*&`, I made some test. 
![image](https://user-images.githubusercontent.com/8220938/127774733-3835a062-ebe2-4b45-a28f-8b5efeb2151c.png)

![image](https://user-images.githubusercontent.com/8220938/127774757-9f8fe630-c128-4fa5-ab97-1ce5f5601a27.png)

looks like the result is not difference.

Reply:
I meant to dereference the pointer, and just store the value in a variable. And then later, store the pointer to that in the config field.

But it doesn't matter anyways since the defer clause is unecessary. So let's remove that.

Reply:
aha, I got it.